Services/DataBaseService/Utils/MigrationEngine.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.SqlClient;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+
+namespace DataBaseService.Utils
+{
+    public class MigrationEngine
+    {
+        private readonly IConfiguration configuration;
+
+        public MigrationEngine(IConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        /// <summary>
+        /// Migration method. 
+        /// Creates "TradingStation" DB with "ExecutedScripts" table, then:
+        /// Runs through all the .sql in the directory (configure in the appsettings.json).
+        /// Checks (by its name) if a script was already executed, executes it, 
+        /// and marks as executed (by adding a row to the table).
+        /// </summary>
+        public void Migrate()
+        {
+            var connectionStringInitial = configuration.GetConnectionString("InitialString");
+            var connectionStringTradingStation = configuration.GetConnectionString("TradingStationString");
+            var scriptsLocation = configuration.GetSection("Locations")["MigrationScripts"];
+            var allScripts = Directory.GetFiles(scriptsLocation, "*.sql");
+            var scriptsToWriteDown = new Dictionary<string, ExecutedScript>
+            {
+                { "CreateDatabase", CreateDatabase(connectionStringInitial) },
+                { "CreateScriptsTable", CreateScriptsTable(connectionStringTradingStation) }
+            };
+            List<string> scriptsToExecute = FilterScriptsToExecute(connectionStringTradingStation, allScripts);
+            string lastScriptToExecute = "[no script selected]";
+
+            try
+            {
+                using (var conn = new SqlConnection(connectionStringTradingStation))
```

Comment:
I'm not sure if `conn` is a good option for a name, but if everyone is ok with that, no problem.

Reply:
Not problem for me